include/swift/Syntax/RawSyntax.h

```diff
@@ -218,9 +221,9 @@ class RawSyntax final
   /// underlying storage.
   /// If \p NodeId is \c None, the next free NodeId is used, if it is passed,
   /// the caller needs to assure that the node ID has not been used yet.
-  RawSyntax(SyntaxKind Kind, ArrayRef<RC<RawSyntax>> Layout, size_t TextLength,
-            SourcePresence Presence, const RC<SyntaxArena> &Arena,
-            llvm::Optional<SyntaxNodeId> NodeId);
+  RawSyntax(SyntaxKind Kind, ArrayRef<const RawSyntax *> Layout,
+            size_t TextLength, SourcePresence Presence,
+            const RC<SyntaxArena> &Arena, llvm::Optional<SyntaxNodeId> NodeId);
```

Comment:
It doesn’t technically have to be. It’s more a design choice. 

I always think of `SyntaxArena` as a ref-counted object and thus, we should write `RC<SyntaxArena>`. `RawSyntax`'s use of `SyntaxArena *` is more like an unowned reference in Swift, which I consider an implementation detail of `RawSyntax`. As such, I decided that all methods on `RawSyntax` should operate on `RC<SyntaxArena>` instead of `SyntaxArena *`. 

This also meant that the API is easier to use since we have isolated the conversion of `RC<SyntaxArena>` -> `SyntaxArena *` to two location instead of having to call `RC<SyntaxArena>::get` whenever invoking one of these functions.

Does this sound good to you or do you disagree?

Reply:
I see. No objection. Thank you :)